### DISCUSSION TITLE
rwkv results/performance?

### DISCUSSION THREAD

**Discussion by orena1** [espnet/espnet#5841] [General]
I saw that you support RWKV. I'm interested in knowing whether you have seen any results from using it in practice. Do you think it is really as good as the papers say it is? I would be interested in seeing any published results that you may have.
Thanks!

**Comment by sw005320** [comment]
@b-flo, can you answer it?

**Comment by b-flo** [comment]
Hi,

>  I'm interested in knowing whether you have seen any results from using it in practice. 

I did some experiments using RWKV as a pred. net. for a Transducer model but I don't have the results at hand, I'll check. But note that the way RWKV is currently defined in ESPnet is not really useful due to the vanilla pred. net. definition and role. It was added (alongside MEGA) for other things I'm working on which are not included here.

Also, [K. An and K. Zhang paper](https://arxiv.org/pdf/2309.14758) may interest you. They experimented with RWKV as an encoder for streaming ASR.

> Do you think it is really as good as the papers say it is? 

I'm not sure what you mean by "as good as the papers say"?
For tasks such as long-form ASR or low resources ASR, RWKVs (or AFTs, MEGAs, etc) have some interesting properties compared to Xformer architectures. It can also useful to address some specific issues such as memory consumption, latency, acoustic/linguistic context-dependencies limitation, etc.

That being said, it's not actively used nor experimented in our field. It could be interesting to do some experiments at larger scale (with other tasks or SR methods for example).

**Reply by orena1** [reply]
Thanks a lot
